packages/@webex/plugin-cc/src/AgentProfile/AgentProfile.ts

```diff
@@ -0,0 +1,108 @@
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.ciUserId = ciUserId;
+    this.agentProfile = {
+      userDetails: {} as UserResponse['body'],
+      agentDesktopProfile: {} as DesktopProfileResponse['body'],
+      teamsList: [] as ListTeamsResponse['body'],
+      auxCodesList: [] as ListAuxCodesResponse['body']['data'],
+    };
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get Agent Profile by providing ciUserId.
+   * @param {string} ciUserId The CI ID of a User.
+   * @returns {Promise<AgentProfileResponse>} A promise that eventually resolves to an API response and return configuration of an Agent.
+   */
+
+  public async getAgentProfile(ciUserId: string): Promise<AgentProfileResponse> {
+    try {
+      const orgId = await this.webex.credentials.getOrgId();
+      if (!ciUserId || !orgId) {
+        this.webex.logger.error(`Please provide ciUserId and orgId`);
+        Promise.reject(new Error('Please provide ciUserId and orgId'));
+      }
+
+      const agentProfileService = new AgentProfileService(
+        ciUserId,
+        orgId,
+        this.webex,
+        this.wccAPIURL
+      );
+
+      const user: UserResponse = await agentProfileService.getUserUsingCI(ciUserId, orgId);
+      this.agentProfile.userDetails = {
+        agentProfileId: user?.body?.agentProfileId || '',
+        teamIds: user?.body?.teamIds || [''],
+        userProfileId: user?.body?.userProfileId || '',
+      };
+
+      const agentDesktopProfile: DesktopProfileResponse =
+        await agentProfileService.retrieveDesktopProfileById(orgId, user?.body?.agentProfileId);
+      this.agentProfile.agentDesktopProfile = {
+        buddyTeams: agentDesktopProfile?.body?.buddyTeams || [''],
+        idleCodes: agentDesktopProfile?.body?.idleCodes || [''],
+        wrapUpCodes: agentDesktopProfile?.body?.wrapUpCodes || [''],
+        loginVoiceOptions: agentDesktopProfile?.body?.loginVoiceOptions || [''],
+        queues: agentDesktopProfile?.body?.queues || [''],
+      };
+      const filter = user?.body?.teamIds;
+      // Call the below two APIs parallel to optimise the Performance.
+      const [teamsList, auxCodesList]: [ListTeamsResponse, ListAuxCodesResponse] =
+        await Promise.all([
+          agentProfileService.getListOfTeams(orgId, page, pageSize, filter, attributes),
+          agentProfileService.getListOfAuxCodes(orgId, page, pageSize, filter, attributes),
+        ]);
+
+      for (const team of teamsList.body) {
+        this.agentProfile.teamsList.push({
+          id: team?.id || '',
+          name: team?.name || '',
+          active: team?.active || false,
+          teamStatus: team?.teamStatus || '',
+          teamType: team?.teamType || '',
+        });
+      }
+
+      for (const auxCode of auxCodesList.body.data) {
+        this.agentProfile.auxCodesList.push({
+          id: auxCode?.id || '',
```

Comment:
Is there a possibility of auxCode not being returned? Should we continue processing in that case? Why are we marking  this optional?

Reply:
same reason as above. Improvements are welcome :)

Reply:
Should the agent be able to log in without auxCodes?